automation_tools/__init__.py

```diff
@@ -1622,6 +1623,30 @@ def setenforce(mode):
     run('setenforce {0}'.format(mode))
 
 
+def host_pings(host):
+    """This ensures the given IP/hostname pings succesfully.
+
+    :param host: A string. The IP or hostname of host.
+
+    """
+    if host is None:
+        print("The hostname/IP is not provided to ping. \
+        Terminating the execution !!")
+        sys.exit(1)
+    while True:
+        command = subprocess.Popen(
+            'ping -c1 {0}'.format(host),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            shell=True
+        )
+        output = command.communicate()[0]
+        if '1 received' in output:
```

Comment:
So in this case if the exit status code anything other than status code 2 means the host is alive.

Where,
0, means the numbers of packets transmitted == no. of packets received, This means the host is alive
1, means the numbers of packets transmitted >= no. of packets received, This also means the host is alive but there is packet loss just because the deadline has reached.
2, means 100% packet loss , the host is not alive at all.

So you think this approach is correct that I should keep on pining the host until the status code is 0/1.